extensions/amp-analytics/0.1/config.js

```diff
@@ -270,6 +271,47 @@ export class AnalyticsConfig {
     });
   }
 
+  /**
+   * Check to see if config has warning, display on console and
+   * remove the key.
+   * @private
+   * @return {!Promise<undefined>}
+   */
+  checkWarningMessage_() {
+    if (this.config_['warningMessage']) {
+      const TAG = this.getName_();
+      const formattedMessage = this.getFormattedWarningMessage_();
+      user().warn(TAG, formattedMessage);
+      delete this.config_['warningMessage'];
+    }
+    return Promise.resolve();
+  }
+
+  /**
+   * Build the formatted warning message that
+   * displays the vendor type and/or url.
+   * Eg: Warning from ./analytics.config.json: I am deprecated
+   * @private
+   * @return {string}
+   */
+  getFormattedWarningMessage_() {
+    const type = this.element_.getAttribute('type');
+
+    let configUrl = type
+      ? this.getVendorUrl_(type)
+      : this.element_.getAttribute('config');
+
+    if (!type && !configUrl) {
+      configUrl = 'vendor config';
+    }
+
+    const message = `Warning from${type ? ` ${type}` : ''}${
+      configUrl ? ` ${configUrl}` : ''
```

Comment:
Let's call it `Warning from analytics vendor ${type}` and `remote config url: ${configUrl}` if exist.

Reply:
Also please do not use `${}`, follow the format warn('warning ..... %s', type).. 
So that the message is static and can be extracted.